src/main/jbake/content/community/docs/developer-guide/alerts.adoc

```diff
@@ -389,3 +451,420 @@ public class Action {
+
+where
+
+[source]
+groupBy(context.accountId)                    Group window events by context "accountId" field
+window(time,10s)                              Define a sliding time window of 10 seconds
+having(count > 1, count.tags.location > 1)    Define an expression on the grouped events
+
+This condition will be true when there are more than 1 events with more than one _location_ tag, so detecting when
+events for the same _accountId_ happens from different places.
+
+The two previous expressions groups all events of the timing window.
+
+We might have scenarios where only specific events should be grouped.
+
+For these cases we can add filters into the expressions like in the following example:
+
+[source]
+event:groupBy(tags.traceId):filter((category == "Credit Check" && text == "Exceptionally Good") || (category == "Stock Check" && text == "Out of Stock")):having(count > 1, count.tags.accountId == 1)
+
+This expression will group events filtered by a expression
+
+[source]
+filter(
+    (category == "Credit Check" && text == "Exceptionally Good") ||
+    (category == "Stock Check" && text == "Out of Stock")
+)
+
+Note that this expression doesn't define an explicit sliding time window, so it will use a default expiration window.
+
+Additional details can be consulted on the JavaDoc of the implementation and examples:
+
+* link:https://github.com/hawkular/hawkular-alerts/blob/master/hawkular-alerts-engine-extensions/hawkular-alerts-events-aggregation/src/main/java/org/hawkular/alerts/extensions/Expression.java[Events Aggregation Expression JavaDoc]
+* link:https://github.com/hawkular/hawkular-alerts/tree/master/examples/events-aggregation[Events Aggregation Expression Examples]
+
+== Elasticsearch Integration
+
+=== Elasticsearch Alerter
+
+The Elasticsearch Alerter will listen for triggers tagged with "Elasticsearch" tag. The Alerter will schedule a
+periodic query to an Elasticsearch system with the info provided from the tagged trigger context. The Alerter
+will convert Elasticsearch documents into Hawkular Alerting Events and send them into the Alerting engine.
+
+The Elasticsearch Alerter uses the following conventions for trigger tags and context:
+
+[cols="^2,10"]
+|=======================
+| *Required* |
+_trigger.tags["Elasticsearch"] = "<any description>"_ +
+ +
+An _Elasticsearch_ tag is required for the alerter to detect this trigger will query to an Elasticsearch system. +
+Value is not necessary, it can be used as a description, it is reserved for future uses. +
+ +
+i.e.
+ +
+trigger.tags["Elasticsearch"] = ""                          // Empty value is valid +
+trigger.tags["Elasticsearch"] = "OpenShift Logging System"  // It can be used as description
+| *Required* |
+_trigger.context["timestamp"] = "<timestamp field>"_ +
+ +
+Documents fetched from Elasticsearch need a date field to indicate the timestamp. +
+This timestamp will be used in the queries to fetch documents in interval basis. +
```

Comment:
I found this section a little confusing. It took me a moment to realize that the value was supposed to be the fieldname in the ES document. Perhaps we should deprecate "timestamp" in favor of context "timestamp.field".  Either way, I suggest the following text:

This defines the timestamp field name for the fetched  Elasticsearch documents.  The timestamp field is used to fetch documents on an interval basis, without overlap.